### DISCUSSION TITLE
Trying to use hook in zustand store

### DISCUSSION THREAD

**Discussion by emimaricic** [pmndrs/zustand#1499] [Q&A]
Hey I'm importing hook from another provider, but there's an error, can anyone help me:
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

```
import create from "zustand";
import { useContract, useProvider, useSigner } from "wagmi";
import AngleDustRaffleABI from "../components/abi/AngleDustRaffleABI.json";

export const useRaffleStore = create((set) => ({
  raffles: [],
  contract: null,
  connectToContract: (signerOrProvider) => {
    set(() => ({
      contract: useContract({
        address: process.env.NEXT_PUBLIC_RAFFLE_CONTRACT_ADDRESS,
        abi: AngleDustRaffleABI.abi,
        signerOrProvider: signerOrProvider,
      }),
    }));
  },
  getRafflesLength: async () => {
    const contractData = await contract.getRafflesLength();
    setRafflesLength(contractData.toNumber());
  },
}));


```

**Comment by dai-shi** [comment]
Zustand store exists outside React. So, you can't use hooks in the store definition.
Usually, a solution is to define a custom hook that uses the store.
